src/renderer/components/map/MapComponent.js

```diff
@@ -50,6 +55,14 @@ class MapComponent extends React.Component {
     this.map.addControl(new mapboxgl.NavigationControl({ showCompass: false }))
   }
 
+  componentWillUnmount () {
+    // save parts of the state we wanna keep
+    const { selectedChat, userid } = this.props
+    let saveState = JSON.parse(JSON.stringify(this.state))
```

Comment:
Why the JSON dance? If you want a copy of the state you can instead do `const saveState = Object.assign({}, this.state)`

Reply:
I wanted a deep copy just to be sure, because I'm deleting things from the state. We probably don't need it at all as the state is destroyed later anyway when I understood react correctly.

Reply:
I removed the copy altogether now